Cython/Compiler/Code.py

```diff
@@ -1712,7 +1754,7 @@ def add_cached_builtin_decl(self, entry):
 
     def put_cached_builtin_init(self, pos, name, cname):
         w = self.parts['cached_builtins']
-        interned_cname = self.get_interned_identifier(name).cname
+        interned_cname = f"{Naming.modulestatevalue_cname}->{self.get_interned_identifier(name).cname}"
         self.use_utility_code(
             UtilityCode.load_cached("GetBuiltinName", "ObjectHandling.c"))
         w.putln('%s = __Pyx_GetBuiltinName(%s); if (!%s) %s' % (
```

Comment:
The f-string result is not the `interned_cname` any more, so I'd either rename the variable or move the pointer prefixing down into the final code building string.